python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +226,68 @@ def __str__(self) -> str:
         return "false"
 
 
+class UnboundIn(BooleanExpression, UnboundPredicate):
+    """IN operation expression"""
+
+    def __init__(self, left: "UnboundReference", right: List[Literal]):  # pylint: disable=super-init-not-called
+        self._left = left
+        self._right = right
+
+    @property
+    def left(self) -> "UnboundReference":
+        return self._left
+
+    @property
+    def right(self) -> List[Literal]:
+        return self._right
+
+    def __eq__(self, other) -> bool:
+        return id(self) == id(other) or (isinstance(other, UnboundIn) and self.left == other.left and self.right == other.right)
+
+    def __invert__(self) -> Not:
+        return Not(self)
```

Comment:
For now, this should throw an exception. The logical inverse of IN is NOT_IN, rather than NOT(IN). When rewriting expressions, it's critical that we don't introduce new `Not` nodes because they can't be evaluated in many cases.

Reply:
now throws exception